src/_pytest/nodes.py

```diff
@@ -91,14 +92,21 @@ def __init__(
         self.parent = parent
 
         #: the pytest config object
-        self.config = config or parent.config
+        if config:
```

Comment:
It's be a bit nicer if all three fields, `config`, `session` and `_nodeid` follow the same pattern. Right now they vary in what is checked first (`is None` or `is not None`), how it's checked (`if x` vs. `if x is not None`) and how the Optional is handled (assert vs. type annotation).

Reply:
this is intended to be slowly broken/fixed after #5975

Reply:
Amended, better/good?